From 8ddc44a847201fb98770b2a891474b7ef65c32b7 Mon Sep 17 00:00:00 2001
From: Omri Itach <omrii@marvell.com>
Date: Sun, 7 Feb 2016 13:14:19 +0200
Subject: [PATCH 0831/1240] fix: eth: register internal NETA ETH units prior to
 external devices

- Ethernet card PCIe E1000 was registed as 1st ETH interface if plugged.
- Env. vars ethaddr & eth1addr are used with the 1st ETH interfaces, accordingly.
- If E1000 PCIe card was pluged, since registered 1st, it uses the 1st MAC
  address var that exits (ethaddr), which was initially configured for
  NETA0 unpredicted behaviour.

This issue fixed by always register NETA ETH units prior to external units,
hence internal NETA units will always use consistend MAC address variables.

Change-Id: I8b8df9cadb8548378307edbd7c16bd09effa8cf1
Signed-off-by: Omri Itach <omrii@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/27268
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
---
 board/mvebu/common/eth_init.c | 16 ++++++++++++----
 1 file changed, 12 insertions(+), 4 deletions(-)

diff --git a/board/mvebu/common/eth_init.c b/board/mvebu/common/eth_init.c
index 5572a46..3b14bcc 100644
--- a/board/mvebu/common/eth_init.c
+++ b/board/mvebu/common/eth_init.c
@@ -24,14 +24,26 @@
 
 DECLARE_GLOBAL_DATA_PTR;
 
+/* This routine must 1st register NETA units, prior to
+ * external ETH interfaces (i.e PCIe USB2ETH, etc).
+ * This requirement must be kept to ensure consistent
+ * alignment with mac address environment variables
+ * ethaddr, eth1addr, etc, are always saved for NETA units
+ */
 int board_eth_init(bd_t *bis)
 {
 
+/* Initialize MDIO driver, for NETA ETH interface */
 #ifdef CONFIG_MVEBU_MDIO
 	mvebu_mdio_initialize(gd->fdt_blob);
 	miiphy_set_current_dev("mvebu_mdio");
 #endif
 
+/* initalize ETH interfaces */
+#ifdef CONFIG_MVNETA
+	cpu_eth_init(bis);
+#endif
+
 #if defined(CONFIG_SK98)
 	skge_initialize(bis);
 #endif
@@ -45,10 +57,6 @@ int board_eth_init(bd_t *bis)
 	enc28j60_initialize(0, 1, 1000, SPI_MODE_0);
 #endif
 
-#ifdef CONFIG_MVNETA
-	cpu_eth_init(bis);
-#endif
-
 #ifdef CONFIG_EEPRO100
 	eepro100_initialize(bis);
 #endif
-- 
1.9.1

